module/scripts/combat/applyDamage.js

```diff
@@ -208,9 +208,12 @@ async function applyDamageToAllLocations(actor, dialogData, damage, totalDamage,
     let locations = actor.getAllLocations().map(location => actor.getLocationObject(location));
 
     let resultPromises = [];
-    locations.forEach(location =>
-        resultPromises.push(calculateDamageWithLocation(actor, dialogData, damage, totalDamage, infoTotalDmg, location))
-    );
+    locations.forEach(location => {
+        const damageDataCopy = Object.assign({}, damage)
+        damageDataCopy.location = location
```

Comment:
Copying should not be necessary due to no write operations happening. Can you elaborate why you added it?

Reply:
Wasn't sure if the `location` of the original damage object is used elsewhere - so decided to make a copies to avoid the original object being modified on each iteration, thus making the final `location` after the cycle being a last array element

Reply:
Before entering this method all adjustment to the damage object should be done. The method should view the damage object as ``final`` in all regards.

Reply:
Sorry now I see what you did there. Your way is actually the correct way and was missing before